src/main/java/org/breedinginsight/brapi/v2/ListController.java

```diff
@@ -0,0 +1,108 @@
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.annotation.*;
+import io.micronaut.security.annotation.Secured;
+import io.micronaut.security.rules.SecurityRule;
+import lombok.extern.slf4j.Slf4j;
+import org.brapi.client.v2.model.exceptions.ApiException;
+import org.brapi.v2.model.core.BrAPIListSummary;
+import org.brapi.v2.model.core.BrAPIListTypes;
+import org.breedinginsight.api.auth.ProgramSecured;
+import org.breedinginsight.api.auth.ProgramSecuredRoleGroup;
+import org.breedinginsight.api.model.v1.request.query.SearchRequest;
+import org.breedinginsight.api.model.v1.response.DataResponse;
+import org.breedinginsight.api.model.v1.response.Response;
+import org.breedinginsight.api.model.v1.validators.QueryValid;
+import org.breedinginsight.brapi.v1.controller.BrapiVersion;
+import org.breedinginsight.brapi.v2.model.request.query.ListQuery;
+import org.breedinginsight.brapi.v2.services.BrAPIListService;
+import org.breedinginsight.brapps.importer.services.ExternalReferenceSource;
+import org.breedinginsight.model.Program;
+import org.breedinginsight.services.ProgramService;
+import org.breedinginsight.services.exceptions.DoesNotExistException;
+import org.breedinginsight.utilities.response.ResponseUtils;
+import org.breedinginsight.utilities.response.mappers.ListQueryMapper;
+
+import javax.inject.Inject;
+import javax.validation.Valid;
+import java.util.List;
+import java.util.UUID;
+
+@Slf4j
+@Controller
+@Secured(SecurityRule.IS_AUTHENTICATED)
+public class ListController {
+
+    private final ProgramService programService;
+    private final BrAPIListService listService;
+    private final ListQueryMapper listQueryMapper;
+
+    @Inject
+    public ListController(ProgramService programService, BrAPIListService listService,
+                          ListQueryMapper listQueryMapper) {
+        this.programService = programService;
+        this.listService = listService;
+        this.listQueryMapper = listQueryMapper;
+    }
+
+    //@Get(BrapiVersion.BRAPI_V2 + "/lists")
+    @Get("/${micronaut.bi.api.version}/programs/{programId}" + BrapiVersion.BRAPI_V2 + "/lists{?queryParams*}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ProgramSecured(roleGroups = {ProgramSecuredRoleGroup.ALL})
+    public HttpResponse<Response<DataResponse<Object>>> getLists(
+            @PathVariable("programId") UUID programId,
+            @QueryValue @QueryValid(using = ListQueryMapper.class) @Valid ListQuery queryParams
+    ) throws DoesNotExistException, ApiException {
+        try {
+            Program program = programService
+                    .getById(programId)
+                    .orElseThrow(() -> new DoesNotExistException("Program does not exist"));
+
+            // get germplasm lists by default
+            BrAPIListTypes type = BrAPIListTypes.fromValue(queryParams.getListType());
+            if (type == null) {
+                type = BrAPIListTypes.GERMPLASM;
+            }
```

Comment:
If no `listType` is supplied it assumes `listType=germplasm` to avoid breaking existing calls for germpalsm lists. Updating the calls and handling the case where `listType=null` are part of [BI-1845](https://breedinginsight.atlassian.net/browse/BI-1845).